### ISSUE TITLE
requesting "all" transcripts only brings back all transcript with genomic descriptions

### ISSUE THREAD

**Issue by peq123** [openvar/variantValidator#223]
@PeteCausey-Freeman 
Hi Pete,
I believe this wasn't the case previously but testing today I noticed that when requesting "all" transcripts
eg
https://rest.variantvalidator.org/VariantValidator/variantvalidator/GRCh38/NM_007294.3%3Ac.5468-13T%3EC/all?content-type=application%2Fjson
only brings back the validation for the mentioned transcript:
{"NM_007294.3:c.5468-13T>C":{"alt_genomic_loci":[],"gene_ids":{"ccds_ids":["CCDS11453","CCDS11454","CCDS11455","CCDS11456","CCDS11459"],"ensembl_gene_id":"ENSG00000012048","entrez_gene_id":"672","hgnc_id":"HGNC:1100","omim_id":["113705"],"ucsc_id":"uc002ict.4"},"gene_symbol":"BRCA1","genome_context_intronic_sequence":"NC_000017.11(NM_007294.3):c.5468-13T>C","hgvs_lrg_transcript_variant":"LRG_292t1:c.5468-13T>C","hgvs_lrg_variant":"LRG_292:g.172169T>C","hgvs_predicted_protein_consequence":{"slr":"NP_009225.1:p.?","tlr":"NP_009225.1(LRG_292p1):p.?"},"hgvs_refseqgene_variant":"NG_005905.2:g.172169T>C","hgvs_transcript_variant":"NM_007294.3:c.5468-13T>C","primary_assembly_loci":{"grch37":{"hgvs_genomic_description":"NC_000017.10:g.41197832A>G","vcf":{"alt":"G","chr":"17","pos":"41197832","ref":"A"}},"grch38":{"hgvs_genomic_description":"NC_000017.11:g.43045815A>G","vcf":{"alt":"G","chr":"17","pos":"43045815","ref":"A"}},"hg19":{"hgvs_genomic_description":"NC_000017.10:g.41197832A>G","vcf":{"alt":"G","chr":"chr17","pos":"41197832","ref":"A"}},"hg38":{"hgvs_genomic_description":"NC_000017.11:g.43045815A>G","vcf":{"alt":"G","chr":"chr17","pos":"43045815","ref":"A"}}},"reference_sequence_records":{"lrg":"http://ftp.ebi.ac.uk/pub/databases/lrgex/LRG_292.xml","protein":"https://www.ncbi.nlm.nih.gov/nuccore/NP_009225.1","refseqgene":"https://www.ncbi.nlm.nih.gov/nuccore/NG_005905.2","transcript":"https://www.ncbi.nlm.nih.gov/nuccore/NM_007294.3"},"refseqgene_context_intronic_sequence":"NG_005905.2(NM_007294.3):c.5468-13T>C","submitted_variant":"NM_007294.3:c.5468-13T>C","transcript_description":"Homo sapiens BRCA1 DNA repair associated (BRCA1), transcript variant 1, mRNA","validation_warnings":[]},"flag":"gene_variant","metadata":{"seqrepo_db":"2018-08-21","uta_schema":"uta_20180821","variantvalidator_hgvs_version":"1.2.5.vv1","variantvalidator_version":"1.0.4.dev17+gd16b9ef.d20200422"}}

but if done with the genomic coordinates, it works as expected:
https://rest.variantvalidator.org/VariantValidator/variantvalidator/GRCh38/NC_000017.11:g.43045815A>G/all?content-type=application%2Fjson

Has it always done this or is this a recent change?

Thanks

**Comment by peq123** [comment]
using vcf style descriptions only brings back a single transcript as well

**Comment by ifokkema** [comment]
> using vcf style descriptions only brings back a single transcript as well

I'm seeing multiple transcripts when sending VCF style variants:
https://rest.variantvalidator.org/VariantValidator/variantvalidator/GRCh38/17-43045815-A-G/all?content-type=application%2Fjson

**Comment by ifokkema** [comment]
> I believe this wasn't the case previously but testing today I noticed that when requesting "all" transcripts (...)
> only brings back the validation for the mentioned transcript:
> 
> but if done with the genomic coordinates, it works as expected: (...)
>
> Has it always done this or is this a recent change?

Looking back in my VV handling code, this behaviour exists for at least 4 months. My code actually relies on VV returning only one variant when I send NM-based coordinates. When sending NM-based descriptions, you get back only that NM. If you'd like all other transcripts, then sending a genomic variant will enable that.

If Pete agrees, I suggest updating the documentation to indicate the `select_transcripts` flag will only be used when sending genome-based coordinates.

**Comment by peq123** [comment]
Looks like I was mistaken by my impression it had been changed. Depending on the variant sources, sometimes the description I send is genomic and afterwards I group the returned data according to genomic coordinates. It is very likely that when I did my initial testing it just happened that the variants I looked at existed in sources with genomic descriptions and had all transcripts brought back so assumed it was the case for all. 

Thanks again
